File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid;
using DevExpress.XtraLayout.Customization;
using DevExpress.XtraGrid.Views.Grid.Customization;
using DevExpress.XtraLayout;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Columns;
using DevExpress.Utils;
using System.Reflection;
using DevExpress.XtraGrid.Views.Base;

namespace WindowsApplication1 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();

        }

        public void InitData() {
            DataTable table = new DataTable();
            table.Columns.Add("BoolProperty", typeof(bool));
            table.Columns.Add("StringProperty", typeof(string));
            table.Columns.Add("CurrentDate", typeof(DateTime));
            table.Columns.Add("IntProperty", typeof(int));
            for (int i = 0; i <= 10; i++) {
                table.Rows.Add(new object[] {  0, i, DateTime.Today, i });
            }
            gridControl1.DataSource = table;
        }

        private void Form1_Load(object sender, EventArgs e) {
            InitData();
          }

        private void gridControl1_EmbeddedNavigator_ButtonClick(object sender, NavigatorButtonClickEventArgs e) {
            GridControl gridControl = (sender as GridControlNavigator).NavigatableControl as GridControl;
            ColumnView view = gridControl.FocusedView as ColumnView;
            if (e.Button.ButtonType == NavigatorButtonType.Custom && view!=null) {
                view.DeleteSelectedRows();
            }
        }


    }
}

File: VB/WindowsApplication1/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid
Imports DevExpress.XtraLayout.Customization
Imports DevExpress.XtraGrid.Views.Grid.Customization
Imports DevExpress.XtraLayout
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.Utils
Imports System.Reflection
Imports DevExpress.XtraGrid.Views.Base

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()

		End Sub

		Public Sub InitData()
			Dim table As New DataTable()
			table.Columns.Add("BoolProperty", GetType(Boolean))
			table.Columns.Add("StringProperty", GetType(String))
			table.Columns.Add("CurrentDate", GetType(DateTime))
			table.Columns.Add("IntProperty", GetType(Integer))
			For i As Integer = 0 To 10
				table.Rows.Add(New Object() { 0, i, DateTime.Today, i })
			Next i
			gridControl1.DataSource = table
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			InitData()
		End Sub

		Private Sub gridControl1_EmbeddedNavigator_ButtonClick(ByVal sender As Object, ByVal e As NavigatorButtonClickEventArgs) Handles gridControl1.EmbeddedNavigator.ButtonClick
			Dim gridControl As GridControl = TryCast((TryCast(sender, GridControlNavigator)).NavigatableControl, GridControl)
			Dim view As ColumnView = TryCast(gridControl.FocusedView, ColumnView)
			If e.Button.ButtonType = NavigatorButtonType.Custom AndAlso view IsNot Nothing Then
				view.DeleteSelectedRows()
			End If
		End Sub


	End Class
End Namespace
